source/loader/ur_libapi.cpp

```diff
@@ -7743,4 +7743,10 @@ ur_result_t UR_APICALL urUsmP2PPeerAccessGetInfoExp(
     return exceptionToResult(std::current_exception());
 }
 
+// This uselessFunc is needed to ensure that pthread is linked correctly against
+// ur_loader.so for gcc-9
+void uselessFunc() {
```

Comment:
Perhaps put this in an annonymous namespace so that it wont conflict with anything?

Reply:
Yep good shout. Change made